src/vent/core.clj

```diff
@@ -184,14 +184,14 @@
           :context (add-context-to implementation context))
 
         (= type :action)
-        (update-in accumulator
-          [:outputs] conj (execute implementation context))
-
-        (= type :piped-action)
-        (let [outcome (execute implementation context)]
+        (let [output (execute implementation context)
+              pipe-output? (and
+                             (map? output)
+                             (-> output meta :ignore not))]
           (-> accumulator
-            (update-in [:outputs] conj outcome)
-            (assoc :context (deep-merge outcome context))))
+            (update-in [:outputs] conj output)
+            (merge (when pipe-output?
+                     {:context (deep-merge context output)}))))
```

Comment:
As we're already ensuring that `output` is a map, the arg order can be as in the gatherer implementation for consistency.